app/assets/javascripts/application.js

```diff
@@ -455,9 +477,60 @@ $(document).on("turbolinks:load", function() {
     });
   }
 
+  // logic to swap genres value from input to search DD and back
+  // this event fires after any of the operators change (in->contains etc)
+  $("#builder").on("afterSetRules.queryBuilder afterUpdateRuleOperator.queryBuilder", function(e, rule) {
+
+    // var name = rule.id;
+    // FIXME: name appears as 'builder_rule_0' because that's the id of the operator DD
+    // We need the name of the value though which is what we have here below so it can be
+    // passed to the backend properly. FIXME: maybe try to find a way to derive this automagically
+    var  name = "builder_rule_0_value_0";
+
+    // only act on genres rule, and 'contains' op
+    if (rule.filter.id == 'genres' && rule.operator.type == 'contains') {
+
+      // use the input method on the filter defined above to get the desired html output
+      var output = rule.filter.input(rule, name)
+
+      // cleanup the prior selectpicker (helps avoid mem leaks)
+      rule.$el.find('.rule-value-container .selectpicker').selectpicker('destroy');
+
+      // render the simple input in the html
+      rule.$el.find('.rule-value-container').html(output)
+
+      // after focus is lost from input, update the rule value so it'll be sent properly to the backend
+      rule.$el.find('.rule-value-container input').one('change', function(e) {
+        rule.value = [this.value]
+      })
+
+      // re-render the fancy multi-select searchable DD
+    } else if (rule.filter.id == 'genres' && rule.operator.type == 'in') {
+
+      // get and render the html
+      var output = rule.filter.input(rule, name)
+      rule.$el.find('.rule-value-container').html(output)
+
+      // force a re-render of the bootstrap-select ui (enhances the <select> with the special features)
+      // https://developer.snapappointments.com/bootstrap-select (<- uses this select DD)
+      rule.$el.find('.rule-value-container select').selectpicker('render');
+
+      // when select DD is hidden / closed, manually assign the selcted values to rule.value
+      // this triggers a model update, which makes the values availible in queryBuilder("getRules")
+      // This is necessary because the automagic 2 way change event binding is missing with our manual setup for genres
+      // https://developer.snapappointments.com/bootstrap-select (<- uses this select DD)
+      rule.$el.find('.rule-value-container select').on('hidden.bs.select', function (e, clickedIndex, isSelected, previousValue) {
+        rule.value = $(this).val()
+      });
```

Comment:
We have to manually listen to input / select changes unfortunately because we're manually generating the input elements, queryBuilder is no longer automatically listening to them to update the rules. This here solves that problem.